src/QirRuntime/public/BitStates.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    /*==============================================================================
+        Provides dynamically extendable storage for packed bits
+    ==============================================================================*/
+    struct BitStates

```

Comment:
Why can't you use `vector<bool>` for this? I think it's pretty equivalent, other than for `IsAny`, and I don't think that method is ever used in a performance-sensitive situation where simply iterating through all of the bits wouldn't be OK.

Reply:
The standard doesn't guarantee that `vector<bool>` is optimized for space.